OMCompiler/Compiler/BackEnd/BackendDAEUtil.mo

```diff
@@ -3074,24 +3093,115 @@ algorithm
     then traversingincidenceRowExpSolvableFinder(e1, tpl);
 
     // use the inlined function to analyze the ocuring variables
-    case (DAE.CALL(path=Absyn.IDENT()), (vars, pa, visitedPaths, ofunctionTree as SOME(functionTree))) guard not AvlSetPath.hasKey(visitedPaths, inExp.path)
+    case (DAE.CALL(path=Absyn.IDENT()), (vars, pa, visitedPaths, isInitial, ofunctionTree as SOME(functionTree))) guard not AvlSetPath.hasKey(visitedPaths, inExp.path)
       algorithm
         (e1,_) := Inline.forceInlineCall(inExp, {}, (SOME(functionTree), {DAE.NORM_INLINE(),DAE.DEFAULT_INLINE()}));
         false := referenceEq(inExp,e1);
-        (_, tpl) := Expression.traverseExpTopDown(e1, traversingincidenceRowExpSolvableFinder, (vars, pa, AvlSetPath.add(visitedPaths, inExp.path), ofunctionTree));
+        (_, tpl) := Expression.traverseExpTopDown(e1, traversingincidenceRowExpSolvableFinder, (vars, pa, AvlSetPath.add(visitedPaths, inExp.path), isInitial, ofunctionTree));
       then (inExp, false, tpl);
 
     else (inExp, true, inTpl);
   end matchcontinue;
 end traversingincidenceRowExpSolvableFinder;
 
+protected function traversingincidenceRowIfExpSolvableFinder
+  "author: kabdelhak FHB 2020-01
+  Checks if condition for special treatment, helper for only traversingincidenceRowExpSolvableFinder.
+      special treatment for inital() condition.
+      Ticket: #5795
+      ToDo: inside more complex expression? IF_EQUATION?"
+  input output DAE.Exp e;
+  output Boolean cont = false; // always false, just for convenience
+  input output tuple<BackendDAE.Variables, AvlSetInt.Tree, AvlSetPath.Tree, Boolean, Option<DAE.FunctionTree>> tpl;
+algorithm
+  tpl := match e
+    local
+      DAE.Exp expCond, expThen, expElse;
+      Boolean isInitial, conditionTrue;
+    case DAE.IFEXP(expCond = expCond, expThen = expThen, expElse = expElse) guard(Expression.containsInitialCall(expCond))
+      algorithm
+        (_, _, _, isInitial, _) := tpl;
```

Comment:
The whole convoluted `isInitial` stuff is only to know if we are analyzing a simulation or an initial system. That information is needed to create the correct entries for expressions like `a = if initial() then b else c;` 

Otherwise unnecessary/ too big / nonlinear  loops can be created. #5795